desktop/core/src/desktop/models.py

```diff
@@ -1855,9 +1855,11 @@ def get_main_quick_action(self, apps):
       LOG.exception('Could not load back default app')
 
     if default_interpreter:
-      return default_interpreter[0]
-    else:
-      return default_app
+      if DEFAULT_INTERPRETER.get() != '':
```

Comment:
If a user stars some other editor, it is handled here or that is taken care in the UI?

Reply:
<img width="847" alt="Screenshot 2023-11-01 at 6 00 06 PM" src="https://github.com/cloudera/hue/assets/68558847/56bf6a7c-d992-4ade-8eec-15fe3ca22885">
`main_button_action` is the variable for the star button on the UI. This is the first priority and then we check for the other 2 conditions.